lib/screens/home/home_view_model.dart

```diff
@@ -0,0 +1,34 @@
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:lydiaryanfluttersurvey/model/response/surveys_response.dart';
+import 'package:lydiaryanfluttersurvey/model/ui/survey_ui_model.dart';
+import 'package:lydiaryanfluttersurvey/screens/home/home_state.dart';
+import 'package:lydiaryanfluttersurvey/usecases/base/base_use_case.dart';
+import 'package:lydiaryanfluttersurvey/usecases/get_surveys_use_case.dart';
+import 'package:rxdart/subjects.dart';
+
+class HomeViewModel extends StateNotifier<HomeState> {
+  final GetSurveysUseCase _getSurveysUseCase;
+
+  final BehaviorSubject<List<SurveyUiModel>> _surveys = BehaviorSubject();
+  Stream<List<SurveyUiModel>> get surveys => _surveys.stream;
+
+  final BehaviorSubject<String> _error = BehaviorSubject();
+  Stream<String> get error => _error.stream;
```

Comment:
we can use `error` from `HomeState` and get rid of this behavior. Tips: We can listen to the update of `HomeState` by calling `ref.listen` ([reference](https://riverpod.dev/docs/concepts/reading#using-reflisten-to-react-to-a-provider-change))

Reply:
Updated in [a57ad5d](https://github.com/nimblehq/flutter-ic-lydia-ryan/pull/79/commits/a57ad5d858ad34ba1c35494ecbac1a8332a45af2) 🙏🏻